flickit-assessment-data/src/main/java/org/flickit/assessment/data/jpa/kit/kitversion/KitVersionJpaRepository.java

```diff
@@ -17,8 +17,11 @@ public interface KitVersionJpaRepository extends JpaRepository<KitVersionJpaEnti
     @Modifying
     @Query("""
             UPDATE KitVersionJpaEntity k
-            SET k.status = :status
+            SET k.status = :status,
+                k.statusVersion = :statusVersion
             WHERE k.id = :id
```

Comment:
**Suggestion:** Ensure that the update query in `updateStatus` method handles potential null values or conflicts when setting `statusVersion`. Consider adding a condition to check the validity of `statusVersion` before the update. [general, importance: 7]
```suggestion
            UPDATE KitVersionJpaEntity k
            SET k.status = :status,
                k.statusVersion = CASE WHEN :statusVersion IS NOT NULL THEN :statusVersion ELSE k.statusVersion END
            WHERE k.id = :id
```